extensions/wikia/WikiaWhiteList/WikiaWhiteList.php

```diff
@@ -93,6 +94,19 @@ function wfWhiteListParse($text) {
 							if (count($m) > 0) {
 								$text = preg_replace(sprintf(REGEX_PARSE_WHITELIST,"rel","nofollow"), "", $text);
 								$text = preg_replace(sprintf(CLASS_PARSE_WHITELIST,"external","(text|free|autonumber)"), "class=\"$1\"", $text);
+
+								$parsed = @parse_url( $lhref );
+								if (!empty( $parsed ) && is_array( $parsed )) {
+									$city_id = WikiFactory::DomainToID(wfNormalizeHost( $parsed['host'] ));
+									if ($city_id) {
+										$parsed['host'] = parse_url(
+											WikiFactory::cityIDtoDomain( $city_id ),
+											PHP_URL_HOST
+										);
+									}
+									$parsed['scheme'] = wfHttpsAllowedForURL( $lhref ) ? 'https' : 'http';
```

Comment:
isn't this part handled by the `onLinkerMakeExternalLink` hook listener? Just asking.